### ISSUE TITLE
Supabase Does Not Provide Access Token To Storage

### ISSUE THREAD

**Issue by jackeverydayzero** [supabase-community/supabase-py#440]
**Describe the bug**
Hey All,
Thanks for the communities work on the Python SDK!!
I recently tried using Supabase storage in the python backend of our app. After a day and a half of debugging, I think I have pegged the issue. It appears that the access token is not being provided to the storage client correctly. 

I have checked that the RLS is correct. I compared this to JS and found this problem.

See below

**To Reproduce**
```
supabase: Client = create_client(
    supabase_url=self.supabaseUrl, supabase_key=self.supabaseAnonKey
)
response = supabase.auth.sign_in_with_password(
    {"email": self.supabaseStandEmail, "password": self.supabaseStandPassword}
)
session = response.session

# Due to a bug detailed here https://github.com/supabase-community/supabase-py/issues/185
# The postgrest client needs to be reinitialized with the auth token
postgrest_client = supabase.postgrest
postgrest_client.auth(session.access_token)

# I had to do this in order to get the storage to make the request correctly.
storageSessionDict = supabase.storage.session.__dict__
storageSessionDict["_headers"]["authorization"] = (
    "Bearer " + supabase.auth.get_session().__dict__["access_token"]
)
```

**Expected behavior**
I expect supabase.storage to just work like the JS SDK.

Thanks for the help! Let me know if i'm missing something.

Jack

**Comment by tobias-scheepers** [comment]
Hi @jackeverydayzero, this is indeed a known issue. See #185 for a workaround until this is fully functional in python.

**Comment by jackeverydayzero** [comment]
@tobias-scheepers I'll be honest mate, that didn't actually fully solve it for me. You can see the code I ended up using to solve it for the moment.
Definitely not ideal, probably doesn't refresh the token probably etc.

```
        print("Signing in to Supabase...")
        response = None
        try:
            response = supabase.auth.sign_in_with_password(
                {
                    "email": self.supabaseStandEmail,
                    "password": self.supabaseStandPassword,
                }
            )
            session = response.session
        except Exception as e:
            raise Exception(
                "Failed to sign in to Supabase. Please check your Supabase credentials.",
                e,
            ) from e

        # Due to a bug detailed here https://github.com/supabase-community/supabase-py/issues/185
        # The postgrest client needs to be reinitialized with the auth token
        postgrest_client = supabase.postgrest
        postgrest_client.auth(session.access_token)

        # It appears that the authorization token is not correctly provided to storage. This of course means that on refresh, this will not be updated.
        storageSessionDict = supabase.storage.session.__dict__
        storageSessionDict["_headers"]["authorization"] = (
            "Bearer " + supabase.auth.get_session().__dict__["access_token"]
        )

```

**Comment by anand2312** [comment]
Fixed by #560
